build.sh

```diff
@@ -121,14 +121,15 @@ else
   $PYTHON_EXECUTABLE -m pip install \
       --target=$ROOT_DIR/python/ray/pyarrow_files pyarrow==0.12.0.RAY \
       --find-links https://s3-us-west-2.amazonaws.com/arrow-wheels/9357dc130789ee42f8181d8724bee1d5d1509060/index.html
-  bazel build //:ray_pkg -c opt --verbose_failures --action_env=PYTHON_BIN_PATH=$PYTHON_EXECUTABLE
-  # Copy files and keep them writeable. This is a workaround, as Bazel
-  # marks all generated files non-writeable. If we would just copy them
-  # over without adding write permission, the copy would fail the next time.
-  # TODO(pcm): It would be great to have a solution here that does not
-  # require us to copy the files.
-  find $ROOT_DIR/bazel-genfiles/ray_pkg/ -exec chmod +w {} \;
-  cp -r $ROOT_DIR/bazel-genfiles/ray_pkg/ray $ROOT_DIR/python || true
+
+  if [ "$RAY_BUILD_JAVA" == "YES" ]; then
+    bazel run //java:bazel_deps -- generate -r $ROOT_DIR -s java/third_party/workspace.bzl -d java/dependencies.yaml
+    bazel build //java:all --verbose_failures
+  fi
+
+  if [ "$RAY_BUILD_PYTHON" == "YES" ]; then
+    bazel build //:ray_pkg --verbose_failures --action_env=PYTHON_BIN_PATH=$PYTHON_EXECUTABLE
```

Comment:
In the past I think we needed -c opt to be able to compile on mac (otherwise there was some clash with a macro names "DEBUG"), but not sure if that's still the case.

Reply:
I think the clash is still there. 
@alegithub111 added `-c opt` to the `.bazelrc`, so we don't need to specify this in every command.

Reply:
Hmm, let's revert this change, we should not assume people are modifying their `.bazelrc` to make this work.

We can also try to find another solution (e.g. remove the clash).

Reply:
@pcmoritz I didn't make myself clear. we added a project-local `.bazelrc` file (see the first file in this PR). People don't need to modify their own files.

Reply:
Got it, great!